XIVComboPlugin/IconReplacer.cs

```diff
@@ -904,6 +916,141 @@ private ulong GetIconDetour(byte self, uint actionID)
                     return actionID;
                 }
             }
+            
+             //Pictomancer
+            if (Configuration.ComboPresets.HasFlag(CustomComboPreset.PictoSubtractivePallet))
+            {
+                if (actionID == PCT.Fire1)
+                {
+                    bool isSubPallet = SearchBuffArray(PCT.SubPallet);
+
+                    if (SearchBuffArray(PCT.Aether1))
+                        return isSubPallet ? PCT.Stone1 : PCT.Aero1;
+
+                    if (SearchBuffArray(PCT.Aether2))
+                        return isSubPallet ? PCT.Thunder1 : PCT.Water1;
+
+                    return isSubPallet ? PCT.Bliz1 : PCT.Fire1;
+                }
+
+                if (actionID == PCT.Fire2)
+                {
+                    bool isSubPallet = SearchBuffArray(PCT.SubPallet);
+
+                    if (SearchBuffArray(PCT.Aether1))
+                        return isSubPallet ? PCT.Stone2 : PCT.Aero2;
+
+                    if (SearchBuffArray(PCT.Aether2))
+                        return isSubPallet ? PCT.Thunder2 : PCT.Water2;
+
+                    return isSubPallet ? PCT.Bliz2 : PCT.Fire2;
+                }
+            }
+
+            if (Configuration.ComboPresets.HasFlag(CustomComboPreset.PictoHolyWhiteCombo))
+            {
+                if (actionID == PCT.HolyWhite)
+                {
+                    if (SearchBuffArray(PCT.Monochrome))
+                        return PCT.CometBlack;
+                    return PCT.HolyWhite;
+                }
+            }
+
+            if (Configuration.ComboPresets.HasFlag(CustomComboPreset.PictoMotifMuseFeature))
+            {
+                var PCTGauge = JobGauges.Get<PCTGauge>();
+                if (actionID == PCT.CreatureMotif)
+                {
+                    if ((PCTGauge.CanvasFlags & CanvasFlags.Pom) != 0)
+                        return PCT.PomMuse;
+
+                    if ((PCTGauge.CanvasFlags & CanvasFlags.Wing) != 0)
+                        return PCT.WingMuse;
+
+                    if ((PCTGauge.CanvasFlags & CanvasFlags.Claw) != 0)
+                        return PCT.ClawMuse;
+
+                    if ((PCTGauge.CanvasFlags & CanvasFlags.Maw) != 0)
+                        return PCT.FangMuse;
+                }
```

Comment:
```suggestion
                    return iconHook.original(self,  actionID);
                }
```

Also, let's move the gauge inside of the ifs, both here and below.